### PR TITLE
use the analysis driver instead of analysis context

### PR DESCRIPTION
**Pull request by jakemac53**

### CODE REVIEW

File: bazel_codegen/lib/src/summaries/resolvers.dart [hunk 1]
```diff
@@ -57,51 +59,49 @@ class SummaryResolvers implements Resolvers {
...
+/// a [Resolver] backed by an [AnalysisDriver].
 class AnalysisResolver implements ReleasableResolver {
-  final AnalysisContext _analysisContext;
+  final AnalysisDriver _analysisDriver;
   final List<AssetId> _assetIds;
 
-  AnalysisResolver(this._analysisContext, this._assetIds);
+  AnalysisResolver(this._analysisDriver, this._assetIds);
 
   @override
```
Comment (natebosch) [inline bazel_codegen/lib/src/summaries/resolvers.dart]: should the annotation be under the doc comment?

File: bazel_codegen/lib/src/summaries/resolvers.dart [hunk 2]
```diff
@@ -57,51 +59,49 @@ class SummaryResolvers implements Resolvers {
...
+  void release() {}
 
   @override
-  bool isLibrary(AssetId assetId) {
+  Future<bool> isLibrary(AssetId assetId) async {
     var uri = assetUri(assetId);
-    var source = _analysisContext.sourceFactory.forUri2(uri);
-    return source != null &&
-        _analysisContext.computeKindOf(source) == SourceKind.LIBRARY;
+    // var source = _analysisDriver.sourceFactory.forUri2(uri);
```
Comment (natebosch) [inline bazel_codegen/lib/src/summaries/resolvers.dart]: delete the commented out code.

Do we need a new way to handle missing sources or will `getSourceKind` not throw anymore?

**Review by natebosch — APPROVED** [review]
Looks good overall.

File: bazel_codegen/pubspec.yaml [hunk 3]
```diff
@@ -5,14 +5,17 @@ homepage: https://github.com/dart-lang/bazel
 version: 0.1.6
 
 environment:
-  sdk: ">=1.8.0 <2.0.0"
+  sdk: ">=1.25.0-dev.7.0 <2.0.0"
```
Comment (natebosch) [inline bazel_codegen/pubspec.yaml]: hmm, we'll probably end up wanting to cut a release that works with build v10 on a non-dev SDK... I'll file an issue.

What are we relying on here?

Comment (natebosch) [inline bazel_codegen/pubspec.yaml]: nevermind - we'd have to cut a new rules_dart and dazel for anyone to pick this up and we can bump versions of the sdk there if we need to.

Would any of these changes be considered breaking?

Comment (jakemac53) [inline bazel_codegen/pubspec.yaml]: This is only for the git path dependencies, I will just revert it actually.

**Review by scheglov — APPROVED** [review]
Analysis Driver changes LGTM.

File: bazel_codegen/lib/src/summaries/resolvers.dart (hunk 1 above)
Comment (jakemac53) [inline bazel_codegen/lib/src/summaries/resolvers.dart]: done, I always want to put it below but the formatter really doesn't like that haha

**Comment by jakemac53** [comment]
Fyi it looks like there will be some more work here so stay tuned for a followup. We are going to have to implement a custom ResourceProvider now it seems like.

**Comment by jakemac53** [comment]
Closing this as there seems to be a lot of other updates that will need to happen since AnalysisDriver doesn't go through the `UriResolver`s when getting library elements or computing source kinds :(.

**Pull request closed without merging**